app/middlewares/credentials-match.js

```diff
@@ -0,0 +1,18 @@
+const bcrypt = require('bcrypt');
+const { conflictError, databaseError, EMAIL_DOES_NOT_EXIST, WRONG_PASSWORD } = require('../errors');
+const { User } = require('../models');
+
+const credentialsMatch = async (req, _res, next) => {
+  try {
+    const user = await User.findOne({ where: { email: req.body.email } });
+    if (user) {
+      const match = await bcrypt.compare(req.body.password, user.password);
+      return match ? next() : next(conflictError('password schema', WRONG_PASSWORD));
```

Comment:
This is not a conflict error, you should create a new error end use `401 Unauthorized`